app/Espo/Core/ORM/Repositories/RDB.php

```diff
@@ -469,6 +469,10 @@ protected function validateVarchar(Entity $entity, string $fieldName, array $fie
             return;
         }
 
+        if($fieldData['disableEmptyValue'] && $entity->get($fieldName) === "") {
```

Comment:
$fieldData['disableEmptyValue'] will returns warning is such key does not exist.